pkg/store/bucket.go

```diff
@@ -716,6 +719,25 @@ func blockSeries(
 			return s.lset[i].Name < s.lset[j].Name
 		})
 
+		validTombstones := make([]tombstone.Tombstone, 0)
+		for _, ts := range tombstones {
+			tsMatchers, err := parser.ParseMetricSelector(ts.Matchers)
```

Comment:
Would be nice avoid extra parsing during query time, instead we preprocess tombstones for quicker querying